src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1667,6 +1693,69 @@ protected override void HandleModulesRequestAsync(IRequestResponder<ModulesArgum
             responder.SetResponse(response);
         }
 
+        // test -- need to delete
+        protected override void HandleDisassembleRequestAsync(IRequestResponder<DisassembleArguments, DisassembleResponse> responder)
+        {
+            DisassembleResponse response = new DisassembleResponse();
+            DisassembleArguments disassembleArguments = responder.Arguments;
+            Debug.Assert(!string.IsNullOrEmpty(disassembleArguments.MemoryReference));
+            try
+            {
+                if (GetMemoryContext(disassembleArguments.MemoryReference, disassembleArguments.Offset, out IDebugMemoryContext2 memoryContext, out ulong address) == HRConstants.S_OK)
+                {
+                    IDebugCodeContext2 codeContext = memoryContext as IDebugCodeContext2;
+                    if (!(codeContext is IDebugCodeContext2))
+                    {
+                        throw new AD7Exception(AD7Resources.Error_ConvertMemoryContext);
+                    }
+                    if (m_program.GetDisassemblyStream(enum_DISASSEMBLY_STREAM_SCOPE.DSS_ALL, codeContext, out IDebugDisassemblyStream2 disassemblyStream) == HRConstants.S_OK)
+                    {
+                        if (disassembleArguments.InstructionOffset != 0)
```

Comment:
According to the spec (https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Disassemble) InstructionOffset is a nullable int. This should probably be `disassembleArguments.InstructionOffset.GetValueOrDefault(0) != 0`

Reply:
I would use `disassembleArguments.InstructionOffset.HasValue` in the conditional and then use `disassembleArguments.InstructionOffset.Value` when using it.